dev-env/docker-compose-dev.yml

```
@@ -0,0 +1,137 @@
+version: '2.4'
+
+services:
+  dev_nginx:
+    container_name: 'dev_nginx_proxy'
+    image: nginx:stable
+    ports:
+      - '8000:80'
+    networks:
+      - dataverse
+    depends_on:
+      - dev_dataverse
+      - dev_frontend
+    volumes:
+      - ./nginx.conf:/etc/nginx/nginx.conf
+      - ./docker-dev-volumes/nginx/logs:/var/log/nginx/
+
+  dev_frontend:
+    container_name: 'dev_frontend'
+    hostname: frontend
+    build:
+      context: ../
+      dockerfile: ./dev.Dockerfile
+      network: host
+    expose:
+      - '5173'
+    stdin_open: true
+    networks:
+      - dataverse
+    depends_on:
+      - dev_dataverse
+    environment:
+      - VITE_DATAVERSE_BACKEND_URL=http://localhost:8000
+    volumes:
+      - ../:/usr/src/app
+      - /usr/src/app/dev-env
+      - /usr/src/app/node_modules
+
+  dev_dataverse:
+    container_name: 'dev_dataverse'
+    hostname: dataverse
+    image: gdcc/dataverse:${DATAVERSE_BRANCH_NAME}
```

May I suggest here that you modify this slightly?
```suggestion
    image: ${REGISTRY}/gdcc/dataverse:${DATAVERSE_BRANCH_NAME}
```

And provide `$REGISTRY` as `docker.io` by default in the `.env`? You will need to switch that to `ghcr.io` to receive the preview images from pull requests.

---

Thanks! Since it is a dev environment and the usual thing will be to pull images of backend feature branches, I think it is more appropriate to set `ghcr.io` as the default registry value.

---

This is really cool, I finally have dataverse running locally 🎉 

I just have a comment about these same lines. 

I tried `./run-env.sh 9444-push-images` but since the branch was merged the script couldn't clone the branch.

Then I ran `./run-env.sh develop` but since the image doesn't exist there was an error.

So I had to change the docker-compose to get the image:

`    image: docker.io/gdcc/dataverse:unstable`

I know there is a line in the README with a disclaimer about this, but what if I want to run the develop branch because the changes in the backend were already merged? Could we provide an easy way to point to develop?

---

Hmm, I tried this:

```
export REGISTRY=docker.io
./run-env.sh unstable
```

But I get this output:

```
INFO - Setting up Dataverse on branch unstable...
INFO - Removing current environment if exists...
Removing network dev-env_dataverse
WARNING: Network dev-env_dataverse not found.
Removing network dev-env_default
WARNING: Network dev-env_default not found.
INFO - Cloning Dataverse backend repository...
Cloning into 'dataverse'...
fatal: Remote branch unstable not found in upstream origin
INFO - Running docker containers...
...
INFO - Bootstrapping dataverse...
./run-env.sh: line 38: ./scripts/dev/docker-final-setup.sh: No such file or directory
INFO - Cleaning up repository...
...
```

So I guess we might need different variables for the branch name to clone (DATAVERSE_BRANCH_NAME) vs. the tag name in the registry (DATAVERSE_IMAGE_TAG?).

For now I'm ok with giving people (us, for now) one of the following workarounds:

- edit the docker compose file
- just use any reasonable tag from https://github.com/orgs/gdcc/packages/container/package/dataverse (though it might be hard for a frontend dev to know which one)

I suspect that for quite a while frontend devs will be pointing at images with new APIs they need, so I'm ok with deferring this until later.

---

The mechanism is closely related to the `ghcr.io` registry and its PR image tags, designed to be used when you want to test a particular SPA feature you are developing that depends on a backend feature branch of Dataverse.

We do not find this relationship between branches names and tags in the `docker.io` registry, so the scripts do not work without applying the modifications you mention.

We can extend the script logic with new variables, as you suggest, to support these cases as well. Anyway, I'm happy to see this approved as it is for now. Let's see what Melina, Ellen or other developers find more useful to improve the mechanism.